src/bridges/set/IssuanceBridge.sol

```diff
@@ -49,150 +72,83 @@ contract IssuanceBridge is IDefiBridge {
         AztecTypes.AztecAsset calldata,
         uint256 inputValue,
         uint256 interactionNonce,
-        uint64,
-        address 
+        uint64 auxData,
+        address
     )
         external
         payable
         override
         returns (
             uint256 outputValueA,
             uint256,
-            bool isAsync
+            bool
         )
     {
-        // Only Rollup Processor can call this function
-        require(
-            msg.sender == rollupProcessor,
-            "IssuanceBridge: INVALID_CALLER"
-        );
-        require(inputValue > 0, "IssuanceBridge: INVALID_INPUT_VALUE");
-
-        isAsync = false;
-
-        // -----------------------------------------------------------------------------------------
-        // Check if the user wants to ISSUE or REDEEM SetToken based on the input/output asset types
-        // -----------------------------------------------------------------------------------------
+        if (msg.sender != rollupProcessor) revert InvalidCaller();
+        if (inputValue == 0) revert ZeroInputValue();
 
-        // REDEEM SET: User wants to redeem SetToken and receive ETH or ERC20
-        // inputAssetA: SetToken
-        // outputAssetA ERC20 or ETH
-        if (
-            setController.isSet(address(inputAssetA.erc20Address)) &&
-            (outputAssetA.assetType == AztecTypes.AztecAssetType.ETH ||
-                outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20)
-        ) {
-            // Check that spending of the given SetToken is approved
-            require(
-                IERC20(inputAssetA.erc20Address).approve(
-                    address(exchangeIssuance),
-                    inputValue
-                ),
-                "IssuanceBridge: APPROVE_FAILED"
-            );
-
-            // user wants to receive ETH
+        if (SET_CONTROLLER.isSet(address(inputAssetA.erc20Address))) {
+            // User wants to redeem the underlying asset
             if (outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
-                outputValueA = exchangeIssuance.redeemExactSetForETH(
+                // redeem in exchange for ETH
+                outputValueA = EXCHANGE_ISSUANCE.redeemExactSetForETH(
                     ISetToken(address(inputAssetA.erc20Address)),
                     inputValue, // _amountSetToken
-                    0 // __minEthOut
+                    auxData // __minEthOut
```

Comment:
There seems to be an extra `_` on the `_minEthOut`. Separately, while it is a good idea to have the constraint, I'm not sure how it should be decided, e.g., if every user uses a different auxData, we cannot do the same aggregation etc.

Reply:
Good point about the lack of aggregation. I think it's still desirable to pass it through auxData. I was thinking that maybe we could simply recommend the front-end implementors to quantize the values (e.g. 0.1 ETH, 1ETH, etc.). I think that in practice everyone will set it to 0 anyway.

Reply:
We've talked in a call with Joe and the decision is that passing it through auxData is fine because frontends are incentivized to aggregate it on their own.